api/admin/oauth.py

```diff
@@ -7,15 +7,13 @@
 
 class GoogleAuthService(object):
 
-    def __init__(self, client_json_file, redirect_uri, test_mode=False):
+    def __init__(self, config, redirect_uri, test_mode=False):
         if test_mode:
             self.client = DummyGoogleClient()
         else:
-            self.client = GoogleClient.flow_from_clientsecrets(
-                client_json_file,
-                scope='https://www.googleapis.com/auth/userinfo.email',
-                redirect_uri=redirect_uri
-            )
+            config['redirect_uri'] = redirect_uri
```

Comment:
Before modifying `config` you should create a new object, i.e. `config = dict(config)`. Otherwise you'll be modifying the global site configuration.